### ISSUE TITLE
Model Data Auto Labeling

### ISSUE THREAD

**Issue by Nrxszv0** [Autonomous-Motorsports-Purdue/AMP_ASS_v8#11]
Description: Use a high performance segmentation model such as SAM3 to automatically label our training data so that we can fine-tune our segmentation model and improve performance. 

Outline:
* Research existing model autolabeling pipeline with SAM2 - talk to Daniel
* Research SAM3 and see how SAM3 could improve the pipeline

Resources:
* https://ai.meta.com/research/sam3/
